### PR TITLE
[ADD] add module stock_backorder_strategy

### PR DESCRIPTION
**Pull request by laetitia-gangloff**
Define on stock_picking_type how to manage backorder.
create : default behavior
no_create : no backorder created
cancel : backorder is cancelled

### CODE REVIEW

File: stock_picking_backorder_strategy/README.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Picking backordering strategies
+===============================
+
+In types of operation add a backorder strategy to be able to cancel created
+backorder or to avoid to create a backorder.
+
+Possible strategies are :
```
Comment (yvaucher) [inline stock_picking_backorder_strategy/README.rst]: Nitpicking of nitpicking: `:` doesn't have a space before it in english. In opposition to french for which it needs to be there.

Same for the 3 next lines

**Comment by yvaucher** [comment]
:+1:

**Comment by laetitia-gangloff** [comment]
What can I do for pylint ?

stock_picking_backorder_strategy/models/stock.py:31: [W0102(dangerous-default-value), StockPicking._create_backorder] Dangerous default value [] as argument (https://travis-ci.org/OCA/stock-logistics-workflow/jobs/75288135#L177)

**Comment by yvaucher** [comment]
@laetitia-gangloff Well I checked it. It would have no side effect as if this argument `backorder_moves` gets it's default mutable value `[]` in official code the variable is check with `if not` and thus it set it to value of a field from picking.

I forgot pylint wouldn't let this pass, so I think you can simply put None instead of [] as default value.

If you want to do more, you can do as we usually did in old API with context.

**Comment by laetitia-gangloff** [comment]
@yvaucher thank you !

I put None, and test it like with context in the old API.

File: stock_picking_backorder_strategy/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    "name": "Picking backordering strategies",
+    "version": "0.1",
+    'author': "Acsone SA/NV, Odoo Community Association (OCA)",
+    "category": "Other",
```
Comment (pedrobaeza) [inline stock_picking_backorder_strategy/__openerp__.py]: Put "Warehouse Management"

**Comment by pedrobaeza** [comment]
One last thing, but :+1:

**Comment by lmignon** [comment]
:+1:  Thank you @laetitia-gangloff

**Pull request merged**